C interface poisson

as4378 opened the pull request:
Hi @tdhock ,

I have created this branch for implementing poisson loss

tdhock commented:
ok great but please lets first focus on merging the gaussian loss/branch
with master

On Tue, Jul 16, 2019 at 7:35 PM Anuraag Srivastava <notifications@github.com>
wrote:

> Hi @tdhock <https://github.com/tdhock> ,
>
> I have created this branch for implementing poisson loss
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/as4378/opart/pull/6
> Commit Summary
>
>    - Added C code for computing optimal changepoints
>    - modified C code
>    - added unit tests for 0 and large penalty
>    - added unit tests for 0 and large penalty
>    - added tests for large and 0 penalty for neuroblastoma and added
>    neuroblastoma.csv for easy access
>    - removed the malloc declarations and passed the intermediate buffers
>    through R code
>    - Added code coverage
>    - Removed function declarations and implemented them directly as they
>    are file specific
>    - Added NEWS file
>    - Removing from this branch and moving to documentation for staging
>    - Added NEWS file
>    - Added informative comments and explained the usage of variables
>    - updated the DESCRIPTION file to use dplyr
>    - temporarily delted neuroblastoma.csv
>    - merged from C_interface
>    - merged the latest C code from C_interface
>    - modified NEWS
>    - Merge branch 'unit_tests' into documentation
>    - Added information about opart in Vignette
>    - removed fpop library
>    - Removed dplyr
>    - modified path
>    - removed beuroblastoma.csv and added dependency on neuroblastoma
>    package
>    - Merge branch 'C_interface' into documentation
>    - modified opart.Rmd to use neuroblastoma package
>    - changed length(data) + 1 to length(data) in R and modified C code
>    accordingly
>    - added test for large number of data points
>    - added run time comparison of opart_gaussian with fpop on
>    neuroblastoma dataset
>    - fixed the issue in C code for F(0) and renamed temporary variable to
>    store cost
>    - Merge branch 'C_interface' into unit_tests
>    - added test for checking correctness of algorithm to return correct
>    cost vector
>    - fixed the declarations in R code for sending cost.vec and end.vec to
>    C code and indexed the columns from name instead of index
>    - Merge branch 'C_interface' into unit_tests
>    - Merge branch 'C_interface' into documentation
>    - added comparison of opart_gaussian with cpt.mean using microbenchmark
>    - added structure for opart_poisson
>    - added squared_sums for checking segment cost
>    - Merge branch 'documentation' into C_interface
>    - added requireNamespace for fpop
>
> File Changes
>
>    - *A* .RData <https://github.com/as4378/opart/pull/6/files#diff-0> (0)
>    - *M* .Rbuildignore
>    <https://github.com/as4378/opart/pull/6/files#diff-1> (1)
>    - *M* DESCRIPTION <https://github.com/as4378/opart/pull/6/files#diff-2>
>    (10)
>    - *A* NEWS <https://github.com/as4378/opart/pull/6/files#diff-3> (9)
>    - *M* R/opart_gaussian.R
>    <https://github.com/as4378/opart/pull/6/files#diff-4> (34)
>    - *A* R/opart_poission.R
>    <https://github.com/as4378/opart/pull/6/files#diff-5> (63)
>    - *A* codecov.yml <https://github.com/as4378/opart/pull/6/files#diff-6>
>    (12)
>    - *A* man/opart_poisson.Rd
>    <https://github.com/as4378/opart/pull/6/files#diff-7> (27)
>    - *M* src/interface.c
>    <https://github.com/as4378/opart/pull/6/files#diff-8> (36)
>    - *M* src/opart_gaussian.c
>    <https://github.com/as4378/opart/pull/6/files#diff-9> (194)
>    - *M* src/opart_gaussian.h
>    <https://github.com/as4378/opart/pull/6/files#diff-10> (3)
>    - *A* src/opart_poisson.c
>    <https://github.com/as4378/opart/pull/6/files#diff-11> (19)
>    - *A* src/opart_poisson.h
>    <https://github.com/as4378/opart/pull/6/files#diff-12> (6)
>    - *M* tests/testthat/test-opart-gaussian.R
>    <https://github.com/as4378/opart/pull/6/files#diff-13> (44)
>    - *M* vignettes/opart.Rmd
>    <https://github.com/as4378/opart/pull/6/files#diff-14> (119)
>
> Patch Links:
>
>    - https://github.com/as4378/opart/pull/6.patch
>    - https://github.com/as4378/opart/pull/6.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/as4378/opart/pull/6?email_source=notifications&email_token=AAHDX4RS53S3JKLFJAZZBULP7YBFLA5CNFSM4IEDJH42YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G7Q756Q>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHDX4RPSG5KECYVMUNQ5L3P7YBFLANCNFSM4IEDJH4Q>
> .
>

as4378 commented:
Sure.

On Tue 16 Jul, 2019, 11:36 PM Toby Dylan Hocking, <notifications@github.com>
wrote:

> ok great but please lets first focus on merging the gaussian loss/branch
> with master
>
> On Tue, Jul 16, 2019 at 7:35 PM Anuraag Srivastava <
> notifications@github.com>
> wrote:
>
> > Hi @tdhock <https://github.com/tdhock> ,
> >
> > I have created this branch for implementing poisson loss
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/as4378/opart/pull/6
> > Commit Summary
> >
> > - Added C code for computing optimal changepoints
> > - modified C code
> > - added unit tests for 0 and large penalty
> > - added unit tests for 0 and large penalty
> > - added tests for large and 0 penalty for neuroblastoma and added
> > neuroblastoma.csv for easy access
> > - removed the malloc declarations and passed the intermediate buffers
> > through R code
> > - Added code coverage
> > - Removed function declarations and implemented them directly as they
> > are file specific
> > - Added NEWS file
> > - Removing from this branch and moving to documentation for staging
> > - Added NEWS file
> > - Added informative comments and explained the usage of variables
> > - updated the DESCRIPTION file to use dplyr
> > - temporarily delted neuroblastoma.csv
> > - merged from C_interface
> > - merged the latest C code from C_interface
> > - modified NEWS
> > - Merge branch 'unit_tests' into documentation
> > - Added information about opart in Vignette
> > - removed fpop library
> > - Removed dplyr
> > - modified path
> > - removed beuroblastoma.csv and added dependency on neuroblastoma
> > package
> > - Merge branch 'C_interface' into documentation
> > - modified opart.Rmd to use neuroblastoma package
> > - changed length(data) + 1 to length(data) in R and modified C code
> > accordingly
> > - added test for large number of data points
> > - added run time comparison of opart_gaussian with fpop on
> > neuroblastoma dataset
> > - fixed the issue in C code for F(0) and renamed temporary variable to
> > store cost
> > - Merge branch 'C_interface' into unit_tests
> > - added test for checking correctness of algorithm to return correct
> > cost vector
> > - fixed the declarations in R code for sending cost.vec and end.vec to
> > C code and indexed the columns from name instead of index
> > - Merge branch 'C_interface' into unit_tests
> > - Merge branch 'C_interface' into documentation
> > - added comparison of opart_gaussian with cpt.mean using microbenchmark
> > - added structure for opart_poisson
> > - added squared_sums for checking segment cost
> > - Merge branch 'documentation' into C_interface
> > - added requireNamespace for fpop
> >
> > File Changes
> >
> > - *A* .RData <https://github.com/as4378/opart/pull/6/files#diff-0> (0)
> > - *M* .Rbuildignore
> > <https://github.com/as4378/opart/pull/6/files#diff-1> (1)
> > - *M* DESCRIPTION <https://github.com/as4378/opart/pull/6/files#diff-2>
> > (10)
> > - *A* NEWS <https://github.com/as4378/opart/pull/6/files#diff-3> (9)
> > - *M* R/opart_gaussian.R
> > <https://github.com/as4378/opart/pull/6/files#diff-4> (34)
> > - *A* R/opart_poission.R
> > <https://github.com/as4378/opart/pull/6/files#diff-5> (63)
> > - *A* codecov.yml <https://github.com/as4378/opart/pull/6/files#diff-6>
> > (12)
> > - *A* man/opart_poisson.Rd
> > <https://github.com/as4378/opart/pull/6/files#diff-7> (27)
> > - *M* src/interface.c
> > <https://github.com/as4378/opart/pull/6/files#diff-8> (36)
> > - *M* src/opart_gaussian.c
> > <https://github.com/as4378/opart/pull/6/files#diff-9> (194)
> > - *M* src/opart_gaussian.h
> > <https://github.com/as4378/opart/pull/6/files#diff-10> (3)
> > - *A* src/opart_poisson.c
> > <https://github.com/as4378/opart/pull/6/files#diff-11> (19)
> > - *A* src/opart_poisson.h
> > <https://github.com/as4378/opart/pull/6/files#diff-12> (6)
> > - *M* tests/testthat/test-opart-gaussian.R
> > <https://github.com/as4378/opart/pull/6/files#diff-13> (44)
> > - *M* vignettes/opart.Rmd
> > <https://github.com/as4378/opart/pull/6/files#diff-14> (119)
> >
> > Patch Links:
> >
> > - https://github.com/as4378/opart/pull/6.patch
> > - https://github.com/as4378/opart/pull/6.diff
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/as4378/opart/pull/6?email_source=notifications&email_token=AAHDX4RS53S3JKLFJAZZBULP7YBFLA5CNFSM4IEDJH42YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G7Q756Q
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAHDX4RPSG5KECYVMUNQ5L3P7YBFLANCNFSM4IEDJH4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/as4378/opart/pull/6?email_source=notifications&email_token=AKR4NULAPZFXYUFRLWHB5JLP7YE2HA5CNFSM4IEDJH42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2BVUWY#issuecomment-511924827>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKR4NUNVW5TUGYMRWEAGHJDP7YE2HANCNFSM4IEDJH4Q>
> .
>

tdhock commented:
usually before merging wth master you should ask for a code review ... next
time please do that.

On Wed, Jul 17, 2019 at 8:57 AM Anuraag Srivastava <notifications@github.com>
wrote:

> Merged #6 <https://github.com/as4378/opart/pull/6> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/as4378/opart/pull/6?email_source=notifications&email_token=AAHDX4VLRVMNRPC7HALDI23P746PNA5CNFSM4IEDJH42YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOSRU5LKY#event-2489963947>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHDX4QBSBSXODM6J3JYWBDP746PNANCNFSM4IEDJH4Q>
> .
>
